pkg/scheduler/preemption/preemption.go

```diff
@@ -167,6 +176,10 @@ func minimalPreemptions(wl *workload.Info, assignment flavorassigner.Assignment,
 		if cq != candCQ && !cqIsBorrowing(candCQ, resPerFlv) {
 			continue
 		}
+		if cq != candCQ && allowBorrowingBelowPriority != nil && priority.Priority(candWl.Obj) >= *allowBorrowingBelowPriority {
+			// we have exceeded the priority threshold, so preemption while borrowing is no longer possible
```

Comment:
Indeed, here I rely on sorting of candidates. The last target (highest priority) is never removed in the second pass (because the look is `for i := len(targets) - 2; i >= 0; i-- {`, so once `allowBorrowing=false`, then it stays.

If the order is random, then potentially the second pass removes the highest priority workload, and `allowBorrowing` can be true again.

If we want to support it, then I can see two options (relying on `allowBorrowingFinal`, and without mutating `allowBorrowing`:
1. have a utility function `allowBorrowingForTargets` which would iterate over the targets, and check if none of them is violating `allowBorrowingBelowPriority`, then `allowBorrowingFinal := allowBorrowing && allowBorrowingForTargets(targets, ...)`
2. keep track of the number of targets that violate `allowBorrowingBelowPriority` (`targetsAboveAllowBorrowingThreshold`., increment the counter in the first pass, and decrement the counter when removing targets in the second pass. Then, `allowBorrowingFinal := allowBorrowing && targetsAboveAllowBorrowingThreshold == 0`

Alternatively, we could keep as is, but just explain better. WDYT?

Reply:
Added a comment for now. If you think the comment is too complex, I'm fine with either (1.) or (2.).